index.js

```diff
@@ -1,8 +1,17 @@
 'use strict';
 var isRoot = require('is-root');
+var defaultUID = {
+	'darwin': 501,
+	'linux': 1000
+};
 
 module.exports = function () {
 	if (isRoot() && process.setuid) {
-		process.setuid(501);
+		var uid = parseInt(process.env.SUDO_UID, 10);
+		if (uid && uid > 0) {
+			process.setuid(uid);
+		} else {
+			process.setuid(defaultUID[process.platform]);
```

Comment:
What default should we use for other platforms?

Windows doesn't count as it doesn't have `process.setuid`, but others like OpenBSD or Solaris.